src/codec.rs

```diff
@@ -1207,13 +1280,20 @@ macro_rules! impl_codec_through_iterator {
 			fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
 				<Compact<u32>>::decode(input).and_then(move |Compact(len)| {
 					input.descend_ref()?;
-					let result = Result::from_iter((0..len).map(|_| Decode::decode(input)));
+					let result = Result::from_iter((0..len).map(|_| {
+						input.on_before_alloc_mem(0 $( + mem::size_of::<$generics>() )*)?;
```

Comment:
Since this is low level runtime code, we need to be extremely paranoid about panics. Please use saturating math.

Reply:
`mem::size_of` is a constant function, so it should already fail at compile time.

Reply:
Changed this. Not applicable anymore.